### PR TITLE
[fix]: [SRM-15594]: fixing gcp logs

### PR DESCRIPTION
**Pull request by deepak-harness**
## Describe your changes
Progress service puts the correlation ids in a cache. But since we are spawning two threads it leads to overwritting the variable. Making it single threaded executor service it will avoid that part. All the other services are suing single threaded progress service only.
Have moved the datacollectioninfo mapper to optional. The will help take care in case there is no base cv config. In case of API call that API call should fail since we don't have correct configuration.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53203)

### CODE REVIEW

**Comment by deepak-harness** [comment]
trigger uts

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/services/impl/CDStageMetaDataServiceImpl.java [hunk 1]
```diff
@@ -29,7 +29,7 @@ public class CDStageMetaDataServiceImpl implements CDStageMetaDataService {
   public ResponseDTO<CDStageMetaDataDTO> getServiceAndEnvironmentRef(YamlNode stageLevelYamlNode) {
     YamlNode pipelineYamlNode = getPipelineYamlNode(stageLevelYamlNode);
     if (Objects.isNull(pipelineYamlNode)) {
-      log.error("Pipeline not found in given Yaml, By passing validation check");
+      log.warn("Pipeline not found in given Yaml, By passing validation check");
```
Comment (Varsha-lalwani) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/services/impl/CDStageMetaDataServiceImpl.java:32]: We should add pipelineYamlNode in the warning message, would be easier to debug why this happened

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/services/impl/CDStageMetaDataServiceImpl.java [hunk 2]
```diff
@@ -68,7 +68,7 @@ private ResponseDTO<CDStageMetaDataDTO> getCdStageMetaDataResponse(String stageI
                                                                                  .pipelineYaml(pipelineYaml)
                                                                                  .build()));
     } catch (Exception e) {
-      log.error("Exception occurred while fetching service and environment reference, Exception: " + e.getMessage());
+      log.warn("Exception occurred while fetching service and environment reference, Exception: " + e.getMessage());
```
Comment (Varsha-lalwani) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/services/impl/CDStageMetaDataServiceImpl.java:71]: same as above

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/entities/PrometheusCVConfig.java [hunk 3]
```diff
@@ -115,7 +115,11 @@ public String getFilters() {
       if (isManualQuery) {
         int startingIndex = query.indexOf('{');
         int closingIndex = query.indexOf('}');
-        return query.substring(startingIndex + 1, closingIndex);
+        if (startingIndex > -1 && closingIndex > -1) {
+          return query.substring(startingIndex + 1, closingIndex);
+        } else {
+          throw new IllegalArgumentException("Query for the prometheus config is incorrect");
```
Comment (Varsha-lalwani) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/entities/PrometheusCVConfig.java:121]: Can we log the query as well.

**Review by Varsha-lalwani — CHANGES_REQUESTED** [review]

**Comment by deepak-harness** [comment]
trigger uts

**Comment by deepak-harness** [comment]
trigger ut5

**Comment by deepak-harness** [comment]
trigger uts

**Comment by deepak-harness** [comment]
trigger checkstylecodeformat

**Pull request merged**